src/main/java/com/sourcegraph/cody/agent/CodyAgentClient.java

```diff
@@ -72,25 +57,14 @@ public CompletableFuture<Void> webviewCreate(WebviewCreateParams params) {
 
   @JsonNotification("webview/postMessage")
   public void webviewPostMessage(WebviewPostMessageParams params) {
-    if (onChatUpdateMessageInProgress != null
-        && params.getMessage().getType().equals(ExtensionMessage.Type.TRANSCRIPT)) {
-      if (Boolean.FALSE.equals(params.getMessage().isMessageInProgress())) {
-        onFinishedProcessing.run();
-      } else if (params.getMessage().getMessages() != null
-          && !params.getMessage().getMessages().isEmpty()) {
-        ApplicationManager.getApplication()
-            .invokeLater(
-                () ->
-                    onChatUpdateMessageInProgress.accept(
-                        Objects.requireNonNull(params.getMessage().getMessages())
-                            .get(params.getMessage().getMessages().size() - 1)));
+    ExtensionMessage extensionMessage = params.getMessage();
 
-      } else {
-        logger.warn("webview/postMessage: no messages in transcript");
-      }
+    if (onNewMessage != null
+        && extensionMessage.getType().equals(ExtensionMessage.Type.TRANSCRIPT)) {
+      ApplicationManager.getApplication().invokeLater(() -> onNewMessage.accept(params));
     } else {
       logger.warn("onChatUpdateMessageInProgress is null or message type is not transcript");
-      logger.warn(String.format("webview/postMessage %s: %s", params.getId(), params.getMessage()));
+      logger.warn(String.format("webview/postMessage %s: %s", params.getId(), extensionMessage));
```

Comment:
I `runIde` for a while. The logging here should not cause any trouble (in the scope of #315). However, we will need to reconsider the handling of other types someday.